python/metatensor-operations/metatensor/operations/sort.py

```diff
@@ -0,0 +1,237 @@
+    >>> # samples are are sorted which are rows in this case
+    >>> block_sorted_stepwise.values
+    array([[5, 3, 4],
+           [8, 6, 7],
+           [2, 0, 1]])
+    >>> sorted_block = metatensor.sort_block(block)
+    >>> np.all(
+    ...     sorted_block.values == block_sorted_stepwise.values
+    ... )  # doing everything at once and verify correctness
+    True
+    >>> # Example for sorting the gradient
+    >>> sorted_block.add_gradient(
+    ...     parameter="g",
+    ...     gradient=TensorBlock(
+    ...         values=np.arange(18).reshape(3, 2, 3),
+    ...         samples=Labels(["sample"], np.array([[1], [2], [0]])),
+    ...         components=[Labels.range("direction", 2)],
+    ...         properties=sorted_block.properties,
+    ...     ),
+    ... )
+    >>> sorted_grad_block = metatensor.sort_block(sorted_block)
+    >>> sorted_grad_block.gradient("g").samples == Labels.range("sample", 3)
+    True
+    >>> sorted_grad_block.gradient("g").properties == sorted_block.properties
+    True
+    >>> sorted_grad_block.gradient(
+    ...     "g"
+    ... ).values  # sorting of components can be clearly seen
+    array([[[12, 13, 14],
+            [15, 16, 17]],
+    <BLANKLINE>
+           [[ 0,  1,  2],
+            [ 3,  4,  5]],
+    <BLANKLINE>
+           [[ 6,  7,  8],
+            [ 9, 10, 11]]])
+
+    """
+    if isinstance(axes, str):
+        if axes == "all":
+            axes_list = ["samples", "components", "properties"]
+        elif axes not in ["samples", "components", "properties"]:
+            raise ValueError(
+                f"input parameter 'axes' may only be one of the strings 'samples', "
+                f"components' or 'properties' but is '{axes}'"
+            )
+        else:
+            axes_list = [axes]
+    elif isinstance(axes, list):
+        for axis in axes:
+            if axis not in ["samples", "components", "properties"]:
+                raise ValueError(
+                    f"input parameter 'axes' may only contain the strings 'samples', "
+                    f"components' or 'properties' but contains '{axis}'"
+                )
+        axes_list = axes
+    else:
+        raise TypeError("input paramater 'axes' should be of type str or list of str")
+
+    result_block = _sort_single_block(block, axes_list, descending)
+
+    # issue #68
+    # for parameter, gradient in block.gradients().items():
+    for parameter, gradient in block.gradients():
+        if len(gradient.gradients_list()) != 0:
+            raise NotImplementedError("gradients of gradients are not supported")
+
+        result_block.add_gradient(
+            parameter=parameter,
+            gradient=_sort_single_block(gradient, axes_list, descending),
+        )
+
+    return result_block
+
+
+def sort(
+    tensor: TensorMap,
+    axes: Union[str, List[str]] = "all",
+    descending: bool = False,
+) -> TensorMap:
+    """
+    Sorts each block separately, see :py:func:`sort_block` for more information
```

Comment:
I would add a short summary of what the function does before this sentence.

Reply:
changed to
```
    Sort the tensor map according to the key values and the blocks for each
    specified axis in :param axes: according to the label values in the axis.

    Each block is sorted separately, see :py:func:`sort_block` for more information
```
----
Is the 
```
:py:func:`sort_block`
```
actually working? I don't know sphinx so well and I saw that you removed it for a class in another place